### PR TITLE
[slack-usergroups] introduce new integration

### PR DESCRIPTION
**Pull request by maorfr**
this integration does:

* reconciles channels for user groups
* reconciles users for user groups

this integration does not:

* create user groups/channels/users
* add/remove users from channels

by using the a user group, members will be added automatically to newly added channels, so the order will be first to update channels and only then to update users.

### CODE REVIEW

File: reconcile/slack_usergroups.py [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+
+    return slack_map
+
+
+def get_current_state(slack_map):
+    current_state = []
+
+    for workspace, spec in slack_map.items():
+        slack = spec['slack']
+        managed_usergroups = spec['manged_usergroups']
```
Comment (jmelis) [inline reconcile/slack_usergroups.py]: typo

File: reconcile/slack_usergroups.py [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+    f = from_state[type]
+    s = subtract_state[type]
+    return [v for k, v in f.items() if k not in s.keys()]
+
+
+def act(desired_state, slack_map):
+    for state in desired_state:
+        workspace = state['workspace']
+        ugid = state['usergroup_id']
+        users = ','.join(state['users'].keys())
```
Comment (jmelis) [inline reconcile/slack_usergroups.py]: can you move this to inside the update_usergroup_users method? I think it's cleaner... passing an array of users sounds much better than a preformatted string.

File: utils/slack_api.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        return {k: v for k, v in self.get('users').items()
+                if k in users_ids}
+
+    def get(self, type):
+        result_key = 'members' if type == 'users' else type
+        results = {}
+        cursor = ''
+
+        if type in self.results:
+            return self.results[type]
```
Comment (jmelis) [inline utils/slack_api.py]: :heart:

Comment (maorfr) [reply utils/slack_api.py]: :heart_decoration:

File: reconcile/slack_usergroups.py (hunk 1 above)
Comment (maorfr) [reply reconcile/slack_usergroups.py]: thanks!

File: reconcile/slack_usergroups.py (hunk 2 above)
Comment (maorfr) [reply reconcile/slack_usergroups.py]: makes sense

**Pull request merged**